pkg/reconciler/ingress/config/istio.go

```diff
@@ -207,8 +231,12 @@ func parseNewFormat(configMap *corev1.ConfigMap) (*Istio, error) {
 		ret.LocalGateways = localGateways
 	}
 
-	if len(ret.LocalGateways) > 1 {
-		return ret, fmt.Errorf("only one local gateway can be defined: current %q value is %q", localGatewaysKey, localGatewaysStr)
+	if len(ret.DefaultExternalGateways()) > 1 {
+		return ret, fmt.Errorf("only one external gateway with no selector can be defined, here: %v", ret.DefaultExternalGateways())
+	}
+
+	if len(ret.DefaultLocalGateways()) > 1 {
```

Comment:
I think we should always ensure the operator configures a default gateway - otherwise there are bunch of error cases we would need to handle in other parts of the code (eg. ingress not matching _any_ gateway)


```suggestion
	if len(ret.DefaultExternalGateways()) != 1  {
		return ret, fmt.Errorf("only one external gateway with no selector can be defined, here: %v", ret.DefaultExternalGateways())
	}

	if len(ret.DefaultLocalGateways()) !== 1 {
```

Reply:
You might have to update your e2e test